### PR TITLE
CONT-100 - Adding pdk compliance

### PR DESCRIPTION
**Pull request by malikparvez**

### CODE REVIEW

File: .fixtures.yml [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
         inifile: "puppetlabs/inifile"
         puppet_authorization: "puppetlabs/puppet_authorization"
+        facts: "puppetlabs/facts"
+    repositories:
+        provision: 'https://github.com/puppetlabs/provision.git'
+        puppet_agent: 
+            repo: 'https://github.com/puppetlabs/puppetlabs-puppet_agent.git'
+            ref: v4.13.0
+    symlinks:
+        node_encrypt: "#{source_dir}"
```
Comment (jordanbreen28) [inline .fixtures.yml:13]: newline needed

File: metadata.json [hunk 2]
```diff
@@ -128,7 +106,12 @@
     }
   ],
   "requirements": [
-    { "name": "puppet",   "version_requirement": ">=6.0.0 < 8.0.0" }
-  ]
+    {
+      "name": "puppet",
+      "version_requirement": ">=6.0.0 < 8.0.0"
```
Comment (jordanbreen28) [inline metadata.json:111]: we should bump this to now include puppet 8/exclude puppet 6.
```suggestion
      "version_requirement": ">=7.0.0 < 9.0.0"
```

File: README.md [hunk 3]
```diff
@@ -308,5 +308,5 @@ validated on anything other than CentOS, though.
 Contact
 -------
 
-binford2k@gmail.com
+BinFord2k@gmail.com
```
[the remark this answers is not in the document]
Comment (malikparvez) [reply README.md]: Removed contact section

File: .fixtures.yml (hunk 1 above)
Comment (malikparvez) [reply .fixtures.yml]: added newline

File: metadata.json (hunk 2 above)
Comment (malikparvez) [reply metadata.json]: changed the version

File: .rubocop.yml [hunk 4]
```diff
@@ -0,0 +1,723 @@
...
+  EnforcedStyle: brackets
+RSpec/MessageSpies:
+  EnforcedStyle: receive
+Style/Documentation:
+  Exclude:
+  - lib/puppet/parser/functions/**/*
+  - spec/**/*
+Style/WordArray:
+  EnforcedStyle: brackets
+Performance/AncestorsInclude:
```
[the remark this answers is not in the document]
Comment (malikparvez) [reply .rubocop.yml:82]: removed 
Performance/BigDecimalWithNumericArgument:
  Enabled: true

[the remark this answers is not in the document]
Comment (jordanbreen28) [reply .rubocop.yml:82]: Thanks @malikparvez but you need to remove all cops without an enforced style

[the remark this answers is not in the document]
Comment (jordanbreen28) [reply .rubocop.yml]: Sorry @malikparvez still isn't correct.. compare with https://github.com/puppetlabs/puppetlabs-postgresql/blob/main/.rubocop.yml

**Comment by jordanbreen28** [comment]
Closing in favour of a new fresh PR.

**Pull request closed without merging**